page.go

```diff
@@ -356,7 +356,10 @@ func (r *optionalPageReader) ReadValues(values []Value) (n int, err error) {
 
 	for n < len(values) && r.offset < len(r.page.definitionLevels) {
 		for n < len(values) && r.offset < len(r.page.definitionLevels) && r.page.definitionLevels[r.offset] != maxDefinitionLevel {
-			values[n] = Value{definitionLevel: r.page.definitionLevels[r.offset]}
+			values[n] = Value{
+				definitionLevel: r.page.definitionLevels[r.offset],
+				columnIndex:     ^int16(r.page.Column()),
```

Comment:
We could move this computation out of the loop, the `Column` call is done on an interface so the compiler won't be able to know the value doesn't change across iterations.

Reply:
Do you prefer it to be in the reader creation or just at the beginning of the function? I'm slightly leaning towards the first just because it prevents more unnecessary calls.

Reply:
I think having it at the beginning of the function is the right balance between not doing unnecessary work and keeping the code simple. 

Caching it in the reader _could_ be the most efficient but it requires somewhat keeping more _global_ state which always make it harder to reason about the code. I'd go for the local variable computed before the loop, which should already amortize the cost to the point that it shouldn't be measurable on any CPU profiles.